### PR TITLE
[GSoC-2020] Experimental Functions - pgr_makeConnected, pgr_isPlanar

### LINKED ISSUE #1366: GSoC '20: Detailed signature of the new function pgr_makeConnected() to be added in pgRouting
### Name of the function:

**pgr_makeConnected()**: Adds the `minimum number of edges` needed to make the input graph connected. The algorithm first identifies all of the connected components in the graph, then adds edges to connect those components together in a path. For example, if a graph contains three connected components A, B, and C, make_connected will add two edges. The two edges added might consist of one connecting a vertex in A with a vertex in B and one connecting a vertex in B with a vertex in C.

### Main Characteristics of the function:

- It will give the minimum list of all edges which are needed in the graph to make the graph connected. 
- Applicable only for undirected graphs.
- The graph can be either weighted or unweighted.
- pgr_makeConnected() Running Time: On a graph with n vertices and m edges, make_connected runs in time O(n + m)

### Variants:

- pgr_makeConnected()

```sql
pgr_makeConnected(edges_sql)
RETURNS SET OF (seq, start_vid, end_vid)
OR EMPTY SET
```
So from nodeA to nodeB there needs to be an edge connected to make the graph connected. So a similar set of these nodes will be returned after the query. This set will be minimum in nature.
### Parameters:

Parameter | Type | Description
-------------- | ------- | ---------------
edges_sql | TEXT | Inner SQL query, as described below.

### Inner Query

**edges_sql**: ​ It should be an SQL query which should return a set of rows with the following columns:

Column | Type | Default | Description
---------- | ------- | --------- | ----------------
id | ANY-INTEGER | | Identifier of the edge
source | ANY-INTEGER | | Identifier of the first end point vertex of the edge
target | ANY-INTEGER | | Identifier of the second end point vertex of the edge
cost | ANY-NUMERICAL | | Weight of the edge (source, target). When negative: edge (source, target) does not exist, therefore it’s not part of the graph.
reverse_cost | ANY-NUMERICAL | -1 | Weight of the edge (target, source). When negative: edge (target, source) does not exist, therefore it’s not part of the graph.

Here,
ANY-INTEGER ​ = ​ SMALLINT, INTEGER, BIGINT
ANY-NUMERICAL ​ = ​ SMALLINT, INTEGER, BIGINT, REAL, FLOAT

### Result Columns:

Returns set of (seq, start_vid, end_vid)

Column | Type | Description
---------- | ------- | --------------
seq | BIGINT | Sequential value starting from ​1
start_vid | BIGINT | Identifier of ​a node in the graph
end_vid | BIGINT | Identifier of a node in the graph

### PR DESCRIPTION
**Pull request by rajhim2**
Closes #1366, #1601.

This PR adds two new pgRouting functions:
- **pgr_makeConnected:** Returns the set of edges that will make the graph connected.
- **pgr_isPlanar:** Returns a boolean depending upon the planarity of the graph. 

This is a final merge PR containing commits from several different PRs combined. 

The intermediate PRs made on the [GSoC-pgRouting](https://github.com/pgRouting/GSoC-pgRouting/) repository are as follows:  

| Description |  Pull Requests |
| -----------| --------------------- |
|pgr_makeConnected |<ul><li>pgRouting/GSoC-pgRouting#52</li><li>pgRouting/GSoC-pgRouting#58</li><li>pgRouting/GSoC-pgRouting#63</li><li>pgRouting/GSoC-pgRouting#66</li></ul> | 
|pgr_isPlanar |<ul><li>pgRouting/GSoC-pgRouting#42</li><li>pgRouting/GSoC-pgRouting#47</li><li>pgRouting/GSoC-pgRouting#49</li><li>pgRouting/GSoC-pgRouting#74</li><li>pgRouting/GSoC-pgRouting#115</li></ul> | 
|Code Review & Documentation |<ul><li>pgRouting/GSoC-pgRouting#119</li><li>pgRouting/GSoC-pgRouting#132</li></ul>

### CODE REVIEW

File: configuration.conf [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#----------------------
+----------------------
```
Comment (dkastl) [inline configuration.conf:1]: I think the `#` got lost here.

File: doc/components/CMakeLists.txt [hunk 2]
```diff
@@ -6,6 +6,7 @@ SET(LOCAL_FILES
 	pgr_biconnectedComponents.rst
 	pgr_articulationPoints.rst
 	pgr_bridges.rst
+    pgr_makeConnected.rst
```
Comment (dkastl) [inline doc/components/CMakeLists.txt:9]: Is indention correct here?

Comment (rajhim2) [reply doc/components/CMakeLists.txt]: https://github.com/pgRouting/pgrouting/pull/1605/commits/0333207386c25011ab386dc22fcdb0b1a6569d21

**Comment by AasheeshT** [comment]
Good work!

**Pull request merged**